nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-administration/src/main/java/org/apache/nifi/admin/service/impl/StandardKeyService.java

```diff
@@ -118,11 +118,20 @@ public void deleteKey(String identity) {
             transaction = transactionBuilder.start();
 
             // delete the keys
-            DeleteKeysAction deleteKeys = new DeleteKeysAction(identity);
-            transaction.execute(deleteKeys);
-
-            // commit the transaction
-            transaction.commit();
+            DeleteKeyAction deleteKey = new DeleteKeyAction(keyId);
+            Integer rowsDeleted = transaction.execute(deleteKey);
+
+            // commit the transaction if we found one and only one matching user identity
+
+            if (rowsDeleted == 0) {
+                rollback(transaction);
+                throw new AdministrationException("Unable to find user key for key ID " + keyId + " to remove token.");
+            } else if (rowsDeleted > 1) {
```

Comment:
The key table has the following schema:
```
CREATE TABLE KEY (
 ID INT NOT NULL PRIMARY KEY AUTO_INCREMENT, 
 IDENTITY VARCHAR2(4096) NOT NULL UNIQUE, 
 KEY VARCHAR2(100) NOT NULL"
)
```
Does `keyId` here refer to the `ID` column in this table?

If so, then there could never be multiple rows with the same ID since that column is the primary key of the table.

Reply:
Yeah this is true, keyId refers to the ID column, whereas previously the delete used the identity which theoretically could have been multiple rows. @mcgilman, was there any specific need for the DeleteKey(s) mechanism to delete multiple rows/identities? See changes to StandardKeyDao.java.